src/reducers/entities/posts.js

```diff
@@ -191,6 +191,42 @@ function handlePendingPosts(pendingPostIds = [], action) {
     }
 }
 
+function handleSendingPosts(sendingPostIds = [], action) {
+    switch (action.type) {
+    case PostTypes.RECEIVED_NEW_POST: {
+        const sendingPostId = action.data.id;
+        if (sendingPostIds.includes(sendingPostId)) {
+            return sendingPostIds;
+        }
+
+        return [
+            ...sendingPostIds,
+            sendingPostId,
+        ];
+    }
+    case PostTypes.RECEIVED_POST: {
+        const sendingPostId = action.data.id;
+        if (!sendingPostIds.includes(sendingPostId)) {
+            return sendingPostIds;
+        }
+
+        return sendingPostIds.filter((postId) => postId !== sendingPostId);
+    }
+    case PostTypes.RECEIVED_POSTS: {
+        const postIds = Object.values(action.data.posts).map((post) => post.pending_post_id);
+
+        const nextSendingPostIds = sendingPostIds.filter((sendingPostId) => !postIds.includes(sendingPostId));
+        if (nextSendingPostIds.length === sendingPostIds.length) {
+            return sendingPostIds;
+        }
+
+        return nextSendingPostIds;
```

Comment:
Maybe directly return:
```
return sendingPostIds.filter((sendingPostId) => !postIds.includes(sendingPostId));
```

Reply:
actually here, I'm with @lieut-data so it does not return a new array every time